### PR TITLE
Topic/migration

### PR DESCRIPTION
**Pull request by Asakiz**

### CODE REVIEW

File: src/agent.cpp [hunk 1]
```diff
@@ -8,85 +8,65 @@
...
+}
 
-  return document.toVariant().toMap();
+QByteArray Agent::abortDownload() {
+  return this->_process_request(QString("update/download/abort"),
+                                QJsonObject());
 }
 
-QVariantMap Agent::doProbe(const QString &serverAddress, bool ignoreProbeASAP) {
+QByteArray Agent::_process_request(const QString &route, QJsonObject json) {
```
Comment (otavio) [inline src/agent.cpp:47]: camelCase.

**Review by otavio — CHANGES_REQUESTED** [review]

File: qmlplugin/agentplugin.hpp [hunk 2]
```diff
@@ -9,16 +9,15 @@
...
-{
-    Q_OBJECT
-    Q_PLUGIN_METADATA(IID QQmlExtensionInterface_iid)
+class AgentPlugin : public QQmlExtensionPlugin {
+  Q_OBJECT
+  Q_PLUGIN_METADATA(IID QQmlExtensionInterface_iid)
 
 public:
-    void registerTypes(const char *uri);
+  void registerTypes(const char *uri);
```
Comment (otavio) [inline qmlplugin/agentplugin.hpp:20]: This seems to belong to the code style fix, no?

**Review by otavio — CHANGES_REQUESTED** [review]

File: qmlplugin/agentplugin.hpp (hunk 2 above)
Comment (Asakiz) [reply qmlplugin/agentplugin.hpp:20]: Yes, i gonna move to the right commit.

File: src/agent.hpp [hunk 3]
```diff
@@ -9,24 +9,25 @@
 #ifndef AGENT_HPP
 #define AGENT_HPP
 
-#include <QObject>
-#include <QUrl>
-#include <QVariantMap>
+#include "qtincludes.hpp"
```
Comment (otavio) [inline src/agent.hpp:12]: Don't include all; include each required header please.

File: src/handler.hpp [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * Copyright (C) 2018
+ * O.S. Systems Sofware LTDA: contato@ossystems.com.br
+ *
+ * SPDX-License-Identifier:     MIT
+ */
+
+#ifndef HANDLER_HPP
+#define HANDLER_HPP
+
+#include "qtincludes.hpp"
```
Comment (otavio) [inline src/handler.hpp:12]: Same here.

**Review by otavio — CHANGES_REQUESTED** [review]

**Review by otavio — APPROVED** [review]

**Pull request merged**